tools/firmware/build_firmware.sh

```diff
@@ -5,22 +5,29 @@ mkdir -p _release
 
 # Variables
 RECOMPILE_BOOTLOADER="false"
-while getopts r flag
+while getopts rv: flag
 do
     case "${flag}" in
         r) RECOMPILE_BOOTLOADER="true";;
+        v) APPLICATION_VERSION=$OPTARG;;
     esac
 done
 
 BOOTLOADER_HEX="_tmp/bootloader.hex"
 
 APPLICATION_HEX_SOURCE="_build/LEKA_V1_2_DEV/app/os/LekaOS.hex"
-APPLICATION_VERSION="1.1.0+0"
 APPLICATION_SIGNED_HEX="_tmp/application-signed.hex"
 
 FIRMWARE_HEX="_release/firmware.hex"
 FIRMWARE_RELEASE="_release/firmware.bin"
 
+APPLICATION_VERSION_REGEX="^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,5}(\+[0-9]+)?$"
```

Comment:
nice! 

I think it would be even better to use the official semver regex to handle all possibles cases:

https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string

https://regex101.com/r/vkijKf/1/

```suggestion
// ? Official semver regex, see
// ? https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
APPLICATION_VERSION_REGEX="^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"

```

Reply:
But it is not compliant with mcuboot and might broke it.

Reply:
ah, in what way is it not compliant?

Reply:
Here a sample of match for this regex:

> 1.0.0-alpha.1
1.0.0-alpha0.valid
1.0.0-alpha.0valid
1.0.0-alpha-a.b-c-somethinglong+build.1-aef.1-its-okay
1.0.0-rc.1+build.1
2.0.0-rc.1+build.123
1.2.3-beta
10.2.3-DEV-SNAPSHOT
1.2.3-SNAPSHOT-123
1.0.0
2.0.0
1.1.7
2.0.0+build.1848
2.0.1-alpha.1227
1.0.0-alpha+beta
1.2.3----RC-SNAPSHOT.12.9.1--.12+788
1.2.3----R-S.12.9.1--.12+meta
1.2.3----RC-SNAPSHOT.12.9.1--.12
1.0.0+0.build.1-rc.10000aaa-kk-0.1
99999999999999999999999.999999999999999999.99999999999999999
1.0.0-0A.is.legal

Some works as `1.0.0` or `2.0.0`, but I already know that `99999999999999999999999.999999999999999999.99999999999999999` will fail since mcuboot cannot handle values above 1 byte for major and minor and 2 bytes for revision (it happened)

For the remaining, I can try them all if you want, but I am pretty sure that at least one will not fit, maybe `1.0.0-alpha-a.b-c-somethinglong+build.1-aef.1-its-okay` or `1.0.0-0A.is.legal`

Reply:
A script to check os version prior to give it to Makefile

Reply:
might by removed if script that check semver **and** mcuboot is accepted